pkg/assembler/graphql/schema/hashEqual.graphql

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2023 The GUAC Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This is experimental and might change in the future!
+
+# Defines a GraphQL schema for the HashEqual. It contains the justification, source, collector, artifact and equal artifact fields
+"""
+HashEqual is an attestation represents when two artifact hash are similar based on a justification.
+
+Justification - string value representing why the artifacts are the equal
+Source - where this attestation was generated from (based on which document)
+Collector - the GUAC collector that collected the document that generated this attestation
+Artifact - the artifact (represented by algorithm and digest) of the pair that are equal
+equalArtifact - the second artifact (represented by algorithm and digest) of the equal pair
+
+"""
+type HashEqual {
+  justification: String!
+  artifact: Artifact!
+  equalArtifact: Artifact!
+  source: String!
+  collector: String!
+}
+
+"""
+HashEqualSpec allows filtering the list of HashEqual to return.
+"""
+input HashEqualSpec {
+  justification: String
+  artifact: ArtifactSpec
+  equalArtifact: ArtifactSpec
+  source: String
+  collector: String
+}
+
+
+extend type Query {
+  "Returns all HashEqual"
+  HashEqual(hashEqualSpec: HashEqualSpec): [HashEqual!]!
+  "Return all equal artifacts based on artifact spec"
```

Comment:
hmm, im a bit hazy on what the use case for this would be, can you share a bit more on what you're thinking?

Reply:
So in terms of the query:
1. Return all `hashEqual` attestations based on the parameters specified. (not sure if we need this?)
2. Return all `hashEqal` attestations for a specific artifact. It can return one or more.